configure.ac

```diff
@@ -19,7 +19,7 @@ AM_SILENT_RULES([yes])
 AC_CANONICAL_HOST
 # Add some default CFLAGS and CXXFLAGS, can be appended to using the environment variables
 CFLAGS="-g -O2 -Wall -Wextra -Wshadow -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls $CFLAGS"
-CXXFLAGS="-g -O2 -Wall -Wextra -Wshadow -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls $CXXFLAGS"
+CXXFLAGS="-std=c++11 -g -O2 -Wall -Wextra -Wshadow -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls $CXXFLAGS"
```

Comment:
isn't this what `AX_CXX_COMPILE_STDCXX_11` is supposed to do/avoid? having both sounds bad to me, but i'm not an autoconf expert.

Reply:
Hmmm, lemme doublecheck the flags actually added.

Reply:
afaik, `AX_CXX_COMPILE_STDCXX_11` checks if the flag is needed or not to support C++11 features. However, if your compiler default is to use C++14, it will happily run the compiler without `-std=c++11`, allowing the (accidental) use of newer C++ features

Reply:
Using the `AX_CXX_COMPILE_STDCXX_11` we know the compiler supports c++11, but there are two issues:

- an extended mode might be used. This can be fixed by using `[std]` instead of `[ext]`.
- Depending on the compiler, it might support a newer version of c++. One of the goals of this commit is to detect (accidental) use of features > c++11. Using  `AX_CXX_COMPILE_STDCXX_11` does not do that.

Reply:
Right. Maybe `AX_CXX_COMPILE_STDCXX_11` can then just go away. (I hope compilers abort if they do not understand `-std=c++11`.